src/test/features/dashboard/claims-state-machine/full-defence-transitions.ts

```diff
@@ -0,0 +1,306 @@
+    })
+  })
+
+  describe('given the claim with full defence made agreement with mediation ', () => {
+    it('should extract the correct state for the claim issued', () => {
+      const claim: Claim = new Claim().deserialize({
+        ...sampleClaimIssueObj,
+        response: {
+          responseType: ResponseType.FULL_DEFENCE,
+          freeMediation: FreeMediationOption.YES
+        },
+        settlement: {
+          partyStatements: [
+            {
+              madeBy: 'CLAIMANT',
+              type: 'ACCEPTATION'
+            },
+            {
+              type: 'OFFER',
+              madeBy: 'DEFENDANT',
+              offer: {
+                content: 'test',
+                completionDate: MomentFactory.parse('2019-05-01')
+              }
+            }
+          ]
+        }
+      })
+
+      let claimState = fullDefenceTransitions(claim)
+      claimState.findState(claimState)
+      expect(claimState.state).to.equal('fd-made-agreement-with-mediation')
+    })
+  })
+
+  describe('given the claim with full defence made agreement without mediation ', () => {
+    it('should extract the correct state for the claim issued', () => {
+      const claim: Claim = new Claim().deserialize({
+        ...sampleClaimIssueObj,
+        response: {
+          responseType: ResponseType.FULL_DEFENCE,
+          freeMediation: FreeMediationOption.NO
+        },
+        settlement: {
+          partyStatements: [
+            {
+              madeBy: 'CLAIMANT',
+              type: 'ACCEPTATION'
+            },
+            {
+              type: 'OFFER',
+              madeBy: 'DEFENDANT',
+              offer: {
+                content: 'test',
+                completionDate: MomentFactory.parse('2019-05-01')
+              }
+            }
+          ]
+        }
+      })
+
+      let claimState = fullDefenceTransitions(claim)
+      claimState.findState(claimState)
+      expect(claimState.state).to.equal('fd-made-agreement-without-mediation')
+    })
+  })
+
+  describe('given the claim with full defence settled with agreement ', () => {
+    it('should extract the correct state for the claim issued', () => {
+      const claim: Claim = new Claim().deserialize({
+        ...sampleClaimIssueObj,
+        response: {
+          responseType: ResponseType.FULL_DEFENCE,
+          freeMediation: FreeMediationOption.NO
+        },
+        settlement: {
+          partyStatements: [
+            {
+              madeBy: 'DEFENDANT',
+              type: 'COUNTERSIGNATURE'
+            },
+            {
+              madeBy: 'CLAIMANT',
+              type: 'ACCEPTATION'
+            },
+            {
+              type: 'OFFER',
+              madeBy: 'DEFENDANT',
+              offer: {
+                content: 'test',
+                completionDate: MomentFactory.parse('2019-05-01')
+              }
+            }
+          ]
+        },
+        settlementReachedAt : MomentFactory.parse('2019-03-01')
```

Comment:
Remove extra space before colon.

Reply:
I will format the file with CMC code style

Reply:
its done, committed the latest changes